src/V3Tristate.cpp

```diff
@@ -636,6 +639,129 @@ class TristateVisitor final : public TristateBaseVisitor {
+    }
+
+    bool assignmentOfValueOnAllBits(AstAssignW* const nodep, bool value) {
+        if (AstConst* const constp = VN_CAST(nodep->rhsp(), Const)) {
+            const V3Number num = constp->num();
+            if (value)
+                return num.isEqAllOnes();
+            else
+                return num.isEqZero();
+        }
+        return false;
+    }
+
+    AstAssignW* getStrongestAssignmentOfValue(const Assigns& assigns, bool value) {
+        auto maxIt = std::max_element(
+            assigns.begin(), assigns.end(), [&](AstAssignW* ap, AstAssignW* bp) {
+                bool valuesOnRhsA = assignmentOfValueOnAllBits(ap, value);
+                bool valuesOnRhsB = assignmentOfValueOnAllBits(bp, value);
+                if (!valuesOnRhsA && !valuesOnRhsB) return false;
+                if (!valuesOnRhsA) return true;
+                if (!valuesOnRhsB) return false;
+                return getStrength(ap, value) < getStrength(bp, value);
+            });
+        // If all assignment don't have const with all bits equal to value on the RHS,
+        // std::max_element will return one of them anyway, so we need to check before returning
+        if (assignmentOfValueOnAllBits(*maxIt, value))
+            return *maxIt;
+        else
+            return nullptr;
+    }
+
+    void removeWeakerAssignments(Assigns& assigns) {
+        // Weaker assignments are these assignments that can't change the final value of the net.
+        // If we know the value of the RHS, we take into account only strength corresponding to its
+        // value.
+
+        // First we want to find strongest assignment that has known value
+        AstAssignW* const strongest0p = getStrongestAssignmentOfValue(assigns, 0);
+        AstAssignW* const strongest1p = getStrongestAssignmentOfValue(assigns, 1);
+        AstAssignW* strongestp = nullptr;
+        uint8_t greatestKnownStrength;
+        if (strongest0p && strongest1p) {
+            strongestp = getStrength(strongest0p, 0) > getStrength(strongest1p, 1) ? strongest0p
+                                                                                   : strongest1p;
+
+            greatestKnownStrength
+                = std::max(getStrength(strongest0p, 0), getStrength(strongest1p, 1));
+        } else if (strongest0p) {
+            strongestp = strongest0p;
+            greatestKnownStrength = getStrength(strongest0p, 0);
+        } else if (strongest1p) {
+            strongestp = strongest1p;
+            greatestKnownStrength = getStrength(strongest1p, 1);
+        }
+
+        // We also want to know the strongest assignment with equal strengths
+        // But we have to skip assignments of z, so I will do it later
+
+        if (strongestp) {
+            // Then we can safely remove all weaker than it
+            // We also remove assignments of the same strength, because we don't need duplicates
+            std::remove_if(assigns.begin(), assigns.end(), [&](AstAssignW* assignp) {
```

Comment:
Note that `std::remove`/`std::remove_if` only shifts elements to the end of the given range, doesn't actually erase them.

Reply:
I must say that I forgot about it. Now it doesn't matter, because I remove these assignments from AST and don't touch the vector after this function, but I will use it in handling more complicated cases. It's better to remove them now, I will commit it in a moment. Thanks.